util/src/main/java/io/camunda/zeebe/util/sched/Actor.java

```diff
@@ -70,7 +72,7 @@ protected void onActorClosing() {
   }
 
   protected void onActorClosed() {
-    // what ever
+    closeFuture.complete(null);
```

Comment:
Actors that override this method are required to call `super.onActorClosed()` to ensure that the future gets completed accordingly.

For example, the `ExporterDirector` overrides `onActorClosed()` and currently does not call `super.onActorClosed()` (and does not have it's own logic to close itself):
https://github.com/camunda-cloud/zeebe/blob/4f4b36922a73a50283c804690cd8af0bdb29890d/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirector.java#L213-L217

So that with this change, the `ExporterDirectorPartitionTransitionStep` would get stuck here and wait for the completion of the future:
https://github.com/camunda-cloud/zeebe/blob/4f4b36922a73a50283c804690cd8af0bdb29890d/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/ExporterDirectorPartitionTransitionStep.java#L40

I think this is also the reason why the build is failing with some test failures: https://ci.zeebe.camunda.cloud/blue/organizations/jenkins/camunda-cloud%2Fzeebe/detail/dd-8547-close-actor-when-failed/2/pipeline/

Also, this change seems to introduce a potential point of failure. Meaning, if someone overwrites the `onActorClosed()` method (and does not implement their own closing mechanism), they always need to ensure that they are calling `super.onActorClosed()`.